build.gradle.kts

```diff
@@ -50,3 +78,79 @@ subprojects {
         }
     }
 }
+
+/**
+ * To run detekt simply:
+ * 1. ./gradlew module:detekt for each module
+ * 2. ./gradlew detekt for whole project
+ */
+fun Project.coreDetektSetup() {
+    // Apply Plugin to sub-project
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+
+    // Configure Detekt
+    detekt {
+        config = files("$rootDir/config/detekt/detekt.yml")
+        buildUponDefaultConfig = true
+        ignoredBuildTypes = listOf("release")
+        source = files(
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_SRC_DIR_JAVA,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_TEST_SRC_DIR_JAVA,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_SRC_DIR_KOTLIN,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_TEST_SRC_DIR_KOTLIN,
+            // Kotlin Multiplatform
+            "src/commonMain/kotlin",
+            "src/androidMain/kotlin",
+            "src/iosMain/kotlin",
+            "src/jvmMain/kotlin",
+            "src/desktopMain/kotlin",
+            "src/jsMain/kotlin",
+        )
+    }
+
+    tasks.withType<Detekt>().configureEach detekt@{
+        exclude("**/build/**", "**/generated/**", "**/resources/**")
+        basePath = rootProject.projectDir.absolutePath
+        autoCorrect = true // Auto corrects common formatting issues
+        // Configure reports here
+        reports {
+            xml.required.set(false)
+            txt.required.set(false)
+            md.required.set(false)
+
+            html {
+                required.set(true)
+                outputLocation.set(
+                    layout.buildDirectory.file("reports/detekt.html")
+                )
+            }
+
+            sarif.required.set(true)
+        }
+
+        // Merged Report
+        finalizedBy(sarifReportMerge)
+        sarifReportMerge.configure {
+            input.from(this@detekt.sarifReportFile)
+        }
+    }
+
+    tasks.withType<DetektCreateBaselineTask>().configureEach detekt@{
+        exclude("**/build/**", "**/generated/**", "**/resources/**")
+        basePath = rootProject.projectDir.absolutePath
+    }
+}
+
+/**
+ * By default Detekt runs when the Gradle :check task is triggered
+ * It should be disabled for now until everything has been correctly setup (including CI)
+ */
+fun Project.filterDetektFromCheckTask() {
+    if (tasks.names.contains("check")) {
+        tasks.named("check").configure {
+            this.setDependsOn(this.dependsOn.filterNot {
+                it is TaskProvider<*> && it.name == "detekt"
+            })
```

Comment:
does this filter out both detekt and ktlint?

Reply:
Only detekt. That will be removed anyways after all the code has been properly formatted and sanitized